### DISCUSSION TITLE
Purpose of _check_methods in lsp utils

### DISCUSSION THREAD

**Discussion by Delta-official** [LazyVim/LazyVim#4351] [Q&A] [answered]
Hi! I'm currently developing my own config for NeoVim and I'm looking at LazyVim since I'm not that experienced with LSP configurations, and I've been wondering what the purpose of [`_check_methods`](https://github.com/LazyVim/LazyVim/blob/12818a6cb499456f4903c5d8e68af43753ebc869/lua/lazyvim/util/lsp.lua#L62-L87) is, considering the fact that [`on_supports_method`](https://github.com/LazyVim/LazyVim/blob/12818a6cb499456f4903c5d8e68af43753ebc869/lua/lazyvim/util/lsp.lua#L107-L119) essentially does the same thing. If anyone could explain it to me, I'd be very grateful!

**Comment by dpetka2001** [comment, marked as answer]
In my limited knowledge as far as I understand it, they're not exactly doing the same thing. `on_supports_method` creates an autocmd that will execute the function that is passed as second argument for the specific method and `_check_methods` just executes the autocmd after evaluating if the client at hand supports the corresponding method.

**Comment by Delta-official** [comment]
That does seem to be the case. Thanks!
